wrapper/src/main/java/software/amazon/jdbc/plugin/customendpoint/CustomEndpointPlugin.java

```diff
@@ -210,22 +204,20 @@ public Connection connect(
    * @return {@link CustomEndpointMonitor}
    */
   protected CustomEndpointMonitor createMonitorIfAbsent(Properties props) {
-    return monitors.computeIfAbsent(
-        this.customEndpointHostSpec.getHost(),
-        (customEndpoint) -> new CustomEndpointMonitorImpl(
-            this.pluginService,
+    return this.monitorService.runIfAbsent(
+        CustomEndpointMonitorImpl.class,
+        this.customEndpointHostSpec.getUrl(),
+        () -> new CustomEndpointMonitorImpl(
+            this.serviceContainer,
```

Comment:
Similar comment

Reply:
See my comment [here](https://github.com/aws/aws-advanced-jdbc-wrapper/pull/1371#discussion_r2064397863), we are using the service container to reduce constructor parameters but I was wondering if there were cases when we want to use the longer constructor parameters for more isolation. Sounds like you prefer monitors to have the longer constructors so that they are more isolated?